CMakeLists.txt

```diff
@@ -13,6 +13,7 @@ include(cmake/SpotifyVersion.cmake)
 include(cmake/OptionalSource.cmake)
 include(cmake/Format.cmake)
 set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE INTERNAL "")
```

Comment:
Is this used somewhere, or is it for debug?

Reply:
With this setting, cmake will export all the make commands to a file called 'compile_commands.json'
This file is used by the clangd or IDE's IntelliSense engine

As this is a cmake standard feature and is a pretty common file, so I consider it is no harm to add it to project CMakeLists.txt

I am more than happy to remove it if you consider it is not appropriate

Reply:
I think it's fine but belongs in its own PR.

Reply:
No problem, I will roll this back and create another pull request